app/controllers/link_controller.rb

```diff
@@ -0,0 +1,63 @@
+require 'net/https'
+require 'uri'
+require 'json'
+require_relative '../../models/link'
+require_relative '../../app/controllers/application_controller'
+require_relative '../../models/category'
+require_relative '../../services/bing_service'
+require 'warden'
+
+class LinkController < ApplicationController
+  bing_search_api = BingService::BingWebSearchApi.new
+
+  get '/category_links/:category_id' do
+    redirect '/login' unless current_user
+    @current_user = current_user
+
+    term = @current_user.categories.find_by(id: params[:category_id].to_i)
+    parsed_json = bing_search_api.links(term)
+
+    @category_links = parsed_json['webPages']['value']
+
+    haml :'category/category_links'
+  end
+
+  get '/category_link/:category_id' do
+    redirect '/login' unless current_user
+    @current_user = current_user
+
+    term = Category.where(id: params[:category_id].to_i).first
+    parsed_json = bing_search_api.links(term)
+
+    result_set = parsed_json['webPages']['value']
+    # result_set.shuffle!
+
+    category = @current_user.categories.find_by(id: params[:category_id].to_i)
+
+    link_name = category.links.find_by(link_name: result_set[0]['displayUrl'])
```

Comment:
Bug: This query has an error. You are finding by `link_name`, yet you are providing a `link`. `link_name` will be` nil` always because the query is incorrect.

See [how you are saving links](https://github.com/JayKay24/micro_learning_web_app/pull/4/files#diff-e50ff0c44e418c0730178bd65a9498e0R46) if you are not convinced